src/js/components/Tip/Tip.js

```diff
@@ -12,78 +13,84 @@ import { Drop } from '../Drop';
+        <span>{children}</span>
+      )) ||
+      Children.only(children);
 
-  const clonedChild = cloneElement(child, {
-    onMouseEnter: (event) => {
-      setOver(true);
-      if (child.props?.onMouseEnter) child.props.onMouseEnter(event);
-    },
-    onMouseLeave: (event) => {
-      setOver(false);
-      if (child.props?.onMouseLeave) child.props.onMouseLeave(event);
-    },
-    onFocus: (event) => {
-      if (usingKeyboard) setOver(true);
-      if (child.props?.onFocus) child.props.onFocus(event);
-    },
-    onBlur: (event) => {
-      if (usingKeyboard) setOver(false);
-      if (child.props?.onBlur) child.props.onBlur(event);
-    },
-    key: 'tip-child',
-    ref: (node) => {
-      // https://github.com/facebook/react/issues/8873#issuecomment-287873307
-      if (typeof componentRef === 'function') {
-        componentRef(node);
-      } else if (componentRef) {
-        // eslint-disable-next-line no-param-reassign
-        componentRef.current = node;
-      }
-      // Call the original ref, if any
-      const { ref: callerRef } = child;
-      if (typeof callerRef === 'function') {
-        callerRef(node);
-      } else if (callerRef) {
-        callerRef.current = node;
-      }
-    },
-  });
+    const clonedChild = cloneElement(child, {
+      onMouseEnter: (event) => {
+        setOver(true);
+        if (child.props?.onMouseEnter) child.props.onMouseEnter(event);
+      },
+      onMouseLeave: (event) => {
+        setOver(false);
+        if (child.props?.onMouseLeave) child.props.onMouseLeave(event);
+      },
+      onFocus: (event) => {
+        if (usingKeyboard) setOver(true);
+        if (child.props?.onFocus) child.props.onFocus(event);
+      },
+      onBlur: (event) => {
+        if (usingKeyboard) setOver(false);
+        if (child.props?.onBlur) child.props.onBlur(event);
+      },
+      key: 'tip-child',
+      ref: (node) => {
+        // https://github.com/facebook/react/issues/8873#issuecomment-287873307
+        if (typeof componentRef === 'function') {
+          componentRef(node);
+        } else if (componentRef) {
+          // eslint-disable-next-line no-param-reassign
+          componentRef.current = node;
+        }
+        // Call the original ref, if any
+        const { ref: callerRef } = child;
+        if (typeof callerRef === 'function') {
+          callerRef(node);
+        } else if (callerRef) {
+          callerRef.current = node;
+        }
+      },
+    });
 
-  return [
-    clonedChild,
-    (over || tooltipOver) && (
-      <Drop
-        target={componentRef.current}
-        trapFocus={false}
-        key="tip-drop"
-        {...theme.tip.drop}
-        {...dropProps}
-        onMouseEnter={() => setTooltipOver(true)}
-        onMouseLeave={() => setTooltipOver(false)}
-      >
-        {plain ? content : <Box {...theme.tip.content}>{content}</Box>}
-      </Drop>
-    ),
-  ];
-});
+    useEffect(() => {
+      setOver(defaultVisible);
+    }, [defaultVisible]);
```

Comment:
Instead of the useEffect could we just have the initial state of `over` set to `defaultVisible`?
For example:
```
const [over, setOver] = useState(defaultVisible);
```

Reply:
ya initially, I tried that approach, but in the Drop component, on the first render, it returned from line 82 because the dropContainer was empty. Therefore, I needed to wait until the child component was mounted, which is why useEffect was used.

![image](https://github.com/grommet/grommet/assets/63608934/3c1ec298-99f2-4166-b4ef-f3c21865f4bd)

Reply:
Gotcha, thanks for the explanation